inference-engine/src/auto_plugin/auto_plugin.hpp

```diff
@@ -41,8 +41,11 @@ class AutoInferencePlugin : public IE::InferencePluginInternal {
         if (GetCore() == nullptr) {
             IE_THROW() << "Please, work with AUTO device via InferencEngine::Core object";
         }
-        auto fullConfig = mergeConfigs(_config, config);
-        auto metaDevices = GetDeviceChoice(fullConfig);
+        {
+            std::lock_guard<std::mutex> lockGuard(_mutex);
+            SetConfig(config);
```

Comment:
you must not change plugin config state during LoadNetwork

Reply:
Originally I add one member _autoConfig, but as I understand, @myshevts prefers reusing _config in the previous discussion:
https://github.com/openvinotoolkit/openvino/pull/5757#discussion_r640479545
https://github.com/openvinotoolkit/openvino/pull/5757#discussion_r640480087 
well, I think we should make agreement.

Reply:
Again, we must not change plugin config state. Only user does this via `ie.SetConfig`
All configs passed to `LoadNetwork` / `QueryNetwork` should be merged with `_config` and used later in the code.

So, revert global config setting and remove mutex.

Reply:
ok, I will revert it. But I think we still need mutex to protect member _autoConfig which stores the configs merged with _config. because we may use the _autoConfig later in any function of AUTO plugin, like GetMetric()->GetOptimizationCapabilities() will use _autoConfig.
BTW, please let me clarify the concern we are facing, since AUTO is trying to cache config during loadNetwork and use it later (in another api function), this is different with all the other plugins, For example, multi-plugin doesn't cache anything, mkldnn_plugin only cache the config from SetConfig() and use it later; So AUTO has multi-functions which could change the config (by SetConfig and LoadNetwork), this behavior is annoying but needed because of this Capabilites check.
Or I am thinking if it is ok to update config only from SetConfig and modify related tests? anyway, I will revert the code firstly, please let me know what's your thought.

Reply:
add @wangleis

Reply:
If you want to want to get limit devices to CPU, GPU only and get their optimizations capabilities via AUTO, just pass:
```
Core core;
auto optCap = core.GetMetric("AUTO:CPU,GPU", METRIC_KEY(OPTIMIZATION_CAPABILITIES));
```
Or if user wants to work with CPU and GPU globally, he can:
```
Core core;
core.SetConfig({ { CONFIG_KEY(AUTO_DEVICE_LIST), "CPU,GPU" } }, "AUTO");
auto optCap = core.GetMetric("AUTO", METRIC_KEY(OPTIMIZATION_CAPABILITIES));
```
No needs to cache anything